src-openmp/decision_tree.cpp

```diff
@@ -384,16 +384,28 @@ double DecisionTree::predict_(DataVector* observation) const
 DataVector DecisionTree::predict(DataFrame* testdata) const
 {
     /** Perform prediction sequentially on each observation and collect a vector of predictions. */
-    DataVector predictions = DataVector(false);  // is_row=false.
+    //DataVector predictions = DataVector(false);  // is_row=false.
+    int n = testdata->length();
+    std::vector<double> preds(n, -1);
     // Make sure tree has been fitted before prediction:
     assert (this->isFitted());
     // Make sure dataframe has the correct number of features (or one extra column with labels).
     assert ( (testdata->width()==this->num_features_) or (testdata->width()==this->num_features_+1) );
-    for (int i = 0; i < testdata->length(); i++)
+    
+    int i;
+    #pragma omp parallel shared(n, preds) private(i)
     {
-        DataVector* observation = testdata->row(i);
-        double prediction = this->predict_(observation);
-        predictions.addValue(prediction);
+        #pragma omp for schedule(dynamic)
+        for (i = 0; i < n; i++)
+        {
+            DataVector* observation = testdata->row(i);
+            double prediction = this->predict_(observation);
+            //predictions.addValue(prediction);
+            preds[i] = prediction;
+        }
```

Comment:
@johannes-kk I think what you did is probably faster actually, since line 389 pre-allocates space (whereas the DataVector does not, which probably adds overhead for dynamic resizing).

If we want to use a DataVector for consistency this approach should work (using setter suggestion in #107):

```suggestion    int n = testdata->length();
    std::vector<double> preds(n, -1);
    DataVector predictions = DataVector(preds,false);  // is_row=false.
    // Make sure tree has been fitted before prediction:
    assert (this->isFitted());
    // Make sure dataframe has the correct number of features (or one extra column with labels).
    assert ( (testdata->width()==this->num_features_) or (testdata->width()==this->num_features_+1) );

    int i;
    #pragma omp parallel shared(n, preds) private(i)
    {
        #pragma omp for schedule(dynamic)
        for (i = 0; i < n; i++)
        {
            DataVector* observation = testdata->row(i);
            double prediction = this->predict_(observation);
            predictions.setValue(i,prediction);
        }
```

Reply:
But looks good!

Reply:
That's a great point on dynamic resizing. I'll keep it at the original commit, but the setters will come in handy elsewhere.